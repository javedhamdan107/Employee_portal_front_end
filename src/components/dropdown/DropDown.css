.dropdown{
    /* width: 30%; */
    width: 80%;
    border-radius: 3px;
    padding: 2%;
    background-color: #fafafa;
    border: 1px solid #d3c9c9;
}
.input-container{
    width: 33%;
    
}
.placeholder{
    color:#868686 ;
}