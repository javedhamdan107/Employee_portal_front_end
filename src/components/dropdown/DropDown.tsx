/* eslint-disable comma-dangle */
import React from "react";
import "./DropDown.css";
export type DropDownPropTypes = {
  options: string[];
  value: string;
  onChange: (e) => void;
  label: string;
};
const DropDown: React.FC<DropDownPropTypes> = ({
  options,
  label,
  value,
  onChange,
}) => {
  return (
    <div className="input-container">
      <label data-testid="label-test">{label}</label>
      <br></br>
      {/* <input className='input'type={type} value={value} onChange={onChange} /> */}
      <select
        className="dropdown"
        onChange={onChange}
        data-testid="select-test"
      >
        {/* <option >Choose {label}</option> */}
        {options.map((ele) => {
          return (
            <option key={ele} selected={value === ele}>
              {ele}
            </option>
          );
        })}
      </select>
    </div>
  );
};

export default DropDown;
