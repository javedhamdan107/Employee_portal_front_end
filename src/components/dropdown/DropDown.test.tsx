import { render, screen, fireEvent } from "@testing-library/react";
import DropDown, { DropDownPropTypes } from "./DropDown";
import '@testing-library/jest-dom'
const options = ["option 1","option 2","option 3"];

describe("If DropDown works properly",()=>{
    test('If label rendered correctly',()=>{
        const DropDownProps:DropDownPropTypes = {
            options:options,
            value:'',
            onChange:()=>{},
            label:'Select An Option'


        }
        render(<DropDown {...DropDownProps}/>)
        const element = screen.getByTestId("label-test");
        expect(element.innerHTML).toBe('Select An Option');

    })

    test('Renders options correctly',()=>{
        const DropDownProps:DropDownPropTypes = {
            options:options,
            value:'',
            onChange:()=>{},
            label:'Select An Option'


        }
        render(<DropDown {...DropDownProps}/>)
        const option1 = screen.getByText("option 1");
        const option2 = screen.getByText("option 2");
        const option3 = screen.getByText("option 3");
        expect(option1).toBeInTheDocument();
        expect(option2).toBeInTheDocument();
        expect(option3).toBeInTheDocument();

    })

    test('If Snapshot exists correctly',()=>{
        const DropDownProps:DropDownPropTypes = {
            options:options,
            value:'',
            onChange:()=>{},
            label:'Select An Option'


        }
        
        const {asFragment } = render(<DropDown {...DropDownProps}/>);
        expect(asFragment).toMatchSnapshot();

    })
    test('calls onChange when an option is selected', () => {
        const onChange = jest.fn();
        const DropDownProps:DropDownPropTypes = {
            options:options,
            value:'',
            onChange,
            label:'Select An Option'


        }
        
        render(<DropDown {...DropDownProps}/>);
    
        const selectElement = screen.getByTestId('select-test');
        fireEvent.change(selectElement, { target: { value: 'option 3' } });
    
        expect(onChange).toBeCalled();
      });

   
   
   
});
