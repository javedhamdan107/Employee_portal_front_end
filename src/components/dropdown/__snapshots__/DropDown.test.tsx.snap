// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Button works properly If Snapshot exists correctly 1`] = `[Function]`;

exports[`If DropDown works properly If Snapshot exists correctly 1`] = `[Function]`;
